### DISCUSSION TITLE
Problems connecting with Python

### DISCUSSION THREAD

**Discussion by stenci** [supabase/supabase#32493] [Questions] [answered]
I have just signed up to supabase, created an organization, a project, a database and a table.

The connection settings show `db.nsqnwzwjjgjfyovodfeb.supabase.co` as the host. I have tried with the PyCharm database connection and it failed.

I am trying to connect with python, but it's telling me:
```
Failed to connect: could not translate host name "db.nsqnwzwjjgjfyovodfeb.supabase.co" to address: Name or service not known
```

I chatted a little with the AI assistant in the supabase dashboard, it asked me to check with telnet, and i got a similar error message:
```
telnet db.nsqnwzwjjgjfyovodfeb.supabase.co 5432
Connecting To db.nsqnwzwjjgjfyovodfeb.supabase.co...Could not open connection to the host, on port 5432: Connect failed
```

The AI assistant suggested other ways to check the connection, and all had similar failure messages.

Can someone please point me to the correct direction so I can do some productive troubleshooting?

(I don't think that sharing the host name is dangerous, please let me know if is, and, if yes, how to change it)

**Comment by GaryAustin1** [comment, marked as answer]
You probably need to use the session pooler mode.  The connection you are using requires ipv6 for the server/os/routers and network.

**Reply by stormsson** [reply]
how do you do that ? 

I don't know if I'm in the same situation , but I have a `nodename nor servname provided, or not known` error

**Reply by GaryAustin1** [reply]
Go to the connections button at top of the dashboard and then look at the connection info for pooler.  You want session mode to replace the direct connection if your environment does not support ipv6.

**Reply by alibassam24** [reply]
I was also getting the same error, and this helped me, Thanks Gary.
What i understood is ipv6 isn't supported on our network/machine, and session pooler uses an ipv4 which works on our network. We can also purchase the ipv4 add on to make direct connection work.

**Comment by stenci** [comment]
Yes, using the session pooler I was able to connect.
Two hours wasted, yay!
